### PR TITLE
GUI: Togglable graphics settings buttons in status bar

### PR DESCRIPTION
**Pull request by CJBok**
image

The buttons are not meant for the casual gamer, but more for the supporters / feedback providers whom test a multitude of games with each yuzu update to spot any fixes / bugs between the 2 render APIs.

- When users provide feedback about graphical issues supported by a screenshot (discord) some most frequently asked questions are then already answered inside the screenshot;
- Interactivity is disabled while ingame;
- #ifdef HAS_VULKAN checks will determine if the API button is interactable or not. (PR will also work on EA builds);

### CODE REVIEW

**Comment by CJBok** [comment]
> The majority of these settings are not supposed to be toggled constantly and are a product of yuzu's GPU development. As we understand the system more we should eventually eliminate much of the complexity of yuzu's graphics settings.
> 
> This PR is not the correct solution to the fact that some games work better with different settings. A proper solution would be further reverse engineering on the system to figure out exactly when to force 30FPS and further debugging to fix async.

Couldn't agree more myself. However, since yuzu is still far away from achieving perfection and there are quit a few supporters out there whom provide feedback and test each available option for every new release, these little buttons will make their contribution just a little bit easier.

I initially wanted to implement these buttons for ASync and Graphics API (OpenGL / Vulkan) only.

Proposition:
- By default the buttons are hidden
- Add a toggle under the settings -> general -> debug tab
- Remove accurate gpu and force 30 fps buttons

**Comment by CJBok** [comment]
- Removed Accurate and 30FPS buttons
- Cleaner UI changes
- Added Vulkan preperations

**Comment by FearlessTobi** [comment]
Render backend can't be changed while ingame and the Vulkan isn't actually in master, so I don't see much use in this button.

Why would I change Async GPU often while ingame?

**Comment by CJBok** [comment]
Hi Tobi,

The buttons are not meant for the casual gamer, but more for the supporters / feedback providers whom test a multitude of games with each yuzu update to spot any fixes / bugs between the 2 render APIs.

- When users provide feedback about graphical issues supported by a screenshot (discord) some most frequently asked questions are then already answered inside the screenshot;
- Interactivity is disabled while ingame;
- #ifdef HAS_VULKAN checks will determine if the API button is interactable or not. (PR will also work on EA builds);

**Comment by bunnei** [comment]
These settings aren't actually toggle-able in-game, right? If so -- I'm OK with this change, so long as it's scoped to just GPU backend and asynch on/off.

**Comment by CJBok** [comment]
> These settings aren't actually toggle-able in-game, right? If so -- I'm OK with this change, so long as it's scoped to just GPU backend and asynch on/off.

they are disabled while in-game but remain visible.

**Comment by FearlessTobi** [comment]
Why does this PR need to change the stylesheet loading logic?

**Comment by CJBok** [comment]
> Why does this PR need to change the stylesheet loading logic?

The current theme / stylesheet logic had 2 different logics in place which was very confusing. And the default theme didn't make use of a stylesheet at all.

Current logic:
- Default = **no stylesheet**
- Dark = **has stylesheet**
- Colorful = Inherent Default theme, **empty stylesheet**, overrides icons
- Colorful Dark = Inherent Dark theme, **no stylesheet**, overrides icons

New logic:
- Default = **has stylesheet**
- Dark = **has stylesheet**
- Colorful = Inherent Default theme, **no stylesheet**, overrides icons
- Colorful Dark = Inherent Dark theme, **no stylesheet**, overrides icons

**Comment by theboy181** [comment]
I would also like to see the docked/undocked icon added to this PR.

**Comment by bunnei** [comment]
@CJBok can you rebase this? Thank you!

File: src/yuzu/main.cpp [hunk 1]
```diff
@@ -2285,8 +2360,16 @@ void GMainWindow::UpdateUITheme() {
     QStringList theme_paths(default_theme_paths);
 
     if (is_default_theme || current_theme.isEmpty()) {
-        qApp->setStyleSheet({});
-        setStyleSheet({});
+        const QString theme_uri(QLatin1Char{':'} + QStringLiteral("default/style.qss"));
```
Comment (lioncash) [inline src/yuzu/main.cpp]: Any reason this isn't:

```suggestion
        const QString theme_uri(QStringLiteral(":default/style.qss"));
```

File: src/yuzu/main.cpp [hunk 2]
```diff
@@ -480,8 +479,72 @@ void GMainWindow::InitializeWidgets() {
...
+        Settings::Apply();
+    });
+    async_status_button->setText(tr("ASYNC"));
+    async_status_button->setCheckable(true);
+    async_status_button->setChecked(Settings::values.use_asynchronous_gpu_emulation);
+    statusBar()->insertPermanentWidget(0, async_status_button);
+
+    // Setup Renderer API button
+    renderer_status_button = new QPushButton();
+    renderer_status_button->setObjectName(tr("RendererStatusBarButton"));
```
Comment (lioncash) [inline src/yuzu/main.cpp]: This likely shouldn't be translatable. This just sets the internal object name (which users wouldn't see anyways).

File: src/yuzu/main.cpp [hunk 3]
```diff
@@ -480,8 +479,72 @@ void GMainWindow::InitializeWidgets() {
...
+        OnDockedModeChanged(!Settings::values.use_docked_mode, Settings::values.use_docked_mode);
+    });
+    dock_status_button->setText(tr("DOCK"));
+    dock_status_button->setCheckable(true);
+    dock_status_button->setChecked(Settings::values.use_docked_mode);
+    statusBar()->insertPermanentWidget(0, dock_status_button);
+
+    // Setup ASync button
+    async_status_button = new QPushButton();
+    async_status_button->setObjectName(tr("TogglableStatusBarButton"));
```
Comment (lioncash) [inline src/yuzu/main.cpp]: This likely shouldn't be translatable. This just sets the internal object name (which users wouldn't see anyways).

File: src/yuzu/main.cpp [hunk 4]
```diff
@@ -480,8 +479,72 @@ void GMainWindow::InitializeWidgets() {
         label->setVisible(false);
         label->setFrameStyle(QFrame::NoFrame);
         label->setContentsMargins(4, 0, 4, 0);
-        statusBar()->addPermanentWidget(label, 0);
+        statusBar()->addPermanentWidget(label);
     }
+
+    // Setup Dock button
+    dock_status_button = new QPushButton();
+    dock_status_button->setObjectName(tr("TogglableStatusBarButton"));
```
Comment (lioncash) [inline src/yuzu/main.cpp]: This likely shouldn't be translatable. This just sets the internal object name (which users wouldn't see anyways).

File: src/yuzu/main.cpp [hunk 5]
```diff
@@ -480,8 +479,72 @@ void GMainWindow::InitializeWidgets() {
...
+    dock_status_button->setCheckable(true);
+    dock_status_button->setChecked(Settings::values.use_docked_mode);
+    statusBar()->insertPermanentWidget(0, dock_status_button);
+
+    // Setup ASync button
+    async_status_button = new QPushButton();
+    async_status_button->setObjectName(tr("TogglableStatusBarButton"));
+    async_status_button->setFocusPolicy(Qt::NoFocus);
+    connect(async_status_button, &QPushButton::clicked, [&] {
+        if (emulation_running)
```
Comment (lioncash) [inline src/yuzu/main.cpp]:
```suggestion
        if (emulation_running) {
            return;
        }
```

File: src/yuzu/main.cpp [hunk 6]
```diff
@@ -480,8 +479,72 @@ void GMainWindow::InitializeWidgets() {
...
+
+#ifndef HAS_VULKAN
+    renderer_status_button->setChecked(false);
+    renderer_status_button->setCheckable(false);
+    renderer_status_button->setDisabled(true);
+#else
+    renderer_status_button->setChecked(Settings::values.renderer_backend ==
+                                       Settings::RendererBackend::Vulkan);
+    connect(renderer_status_button, &QPushButton::clicked, [=] {
+        if (emulation_running)
```
Comment (lioncash) [inline src/yuzu/main.cpp]:
```suggestion
        if (emulation_running) {
            return;
        }
```

File: src/yuzu/main.cpp [hunk 7]
```diff
@@ -1831,6 +1900,13 @@ void GMainWindow::OnConfigure() {
     }
 
     config->Save();
+
+    dock_status_button->setChecked(Settings::values.use_docked_mode);
+    async_status_button->setChecked(Settings::values.use_asynchronous_gpu_emulation);
+#ifdef HAS_VULKAN
+    renderer_status_button->setChecked(
+        Settings::values.renderer_backend == Settings::RendererBackend::Vulkan ? true : false);
```
Comment (lioncash) [inline src/yuzu/main.cpp]:
```suggestion
        Settings::values.renderer_backend == Settings::RendererBackend::Vulkan);
```

File: src/yuzu/main.cpp [hunk 8]
```diff
@@ -480,8 +479,72 @@ void GMainWindow::InitializeWidgets() {
...
+    async_status_button->setChecked(Settings::values.use_asynchronous_gpu_emulation);
+    statusBar()->insertPermanentWidget(0, async_status_button);
+
+    // Setup Renderer API button
+    renderer_status_button = new QPushButton();
+    renderer_status_button->setObjectName(tr("RendererStatusBarButton"));
+    renderer_status_button->setCheckable(true);
+    renderer_status_button->setFocusPolicy(Qt::NoFocus);
+    connect(renderer_status_button, &QPushButton::toggled, [=](bool checked) {
+        renderer_status_button->setText(tr(checked ? "VULKAN" : "OPENGL"));
```
Comment (lioncash) [inline src/yuzu/main.cpp]:
```suggestion
        renderer_status_button->setText(checked ? tr("VULKAN") : tr("OPENGL"));
```

**Comment by FearlessTobi** [comment]
Seems like you messed up your rebase.

**Comment by CJBok** [comment]
> Seems like you messed up your rebase.

just a little... >.<

**Comment by CJBok** [comment]
Will raise new PR with corrections.

**Pull request closed without merging**